util/lua/lua.go

```diff
@@ -290,7 +290,8 @@ func cleanReturnedObj(newObj, obj map[string]any) map[string]any {
 				switch oldValue := oldValueInterface.(type) {
 				case map[string]any:
 					if len(newValue) == 0 {
-						mapToReturn[key] = oldValue
+						// Lua incorrectly decoded the empty object as an empty array, so set it to an empty object
+						mapToReturn[key] = map[string]any{}
```

Comment:
Can this failure/fix be unit tested?

My reading of the original code is that it's impossible for `oldValue` to be an array, because we're in a case that asserts that the type is `map[string]any{}`.

Reply:
I assume the new Lua test case covers this, but that test doesn't communicate the intent of the behavior. I think a test saying "here's how we expect cleaning to work given X data" will be easier to understand and maintain.

Reply:
Correct, we had to read the full code to understand that oldValue will be `map[string]any{}` 100% of the time. This clarifies it.